beacon-chain/rpc/validator/proposer.go

```diff
@@ -664,9 +665,15 @@ func (vs *Server) packAttestations(ctx context.Context, latestState *stateTrie.B
 		uAtts := vs.AttPool.UnaggregatedAttestations()
 		uAtts, err = vs.filterAttestationsForBlockInclusion(ctx, latestState, uAtts)
 		numUAtts := uint64(len(uAtts))
+		uAtts, err = attaggregation.Aggregate(uAtts)
+		if err != nil {
+			return nil, err
+		}
+
 		if numUAtts+numAtts > params.BeaconConfig().MaxAttestations {
 			uAtts = uAtts[:params.BeaconConfig().MaxAttestations-numAtts]
 		}
+
 		atts = append(atts, uAtts...)
```

Comment:
```suggestion
		uAtts, err = attaggregation.Aggregate(uAtts)
		if err != nil {
			return nil, err
		}

		atts = append(atts, uAtts...)
		atts, err = attaggregation.Aggregate(atts)
		if err != nil {
		   return nil, err
		}
		if len(atts) > params.BeaconConfig().MaxAttestations {
		   atts = sort.MostProfitable(atts)
		   atts = atts[:params.BeaconConfig().MaxAttestations]
		}
		atts = sort.MostProfitable(atts)
```

What about something like this?

Reply:
Much better

Reply:
The sort thing is not implemented though!